api/app/Models/PoolCandidate.php

```diff
@@ -280,19 +303,44 @@ public function filterByEquity(Builder $query, array $equity): Builder
         };
 
         // then return queries depending on above array count, special query syntax needed for multiple ORs to ensure proper SQL query formed
-        $query->where(function ($query) use ($equityVars) {
-            foreach ($equityVars as $index => $equityInstance) {
-                if ($index === 0) {
-                    // First iteration must use where instead of orWhere, as seen in filterWorkRegions
-                    $query->where($equityVars[$index], true);
-                } else {
-                    $query->orWhere($equityVars[$index], true);
-                }
-            }
+        $query->whereExists(function ($query) use ($equityVars) {
+            $query->select('id')
+                ->from('users')
+                ->whereColumn('users.id', 'pool_candidates.user_id')
+                ->where(function ($query) use ($equityVars) {
+                    foreach ($equityVars as $index => $equityInstance) {
+                        if ($index === 0) {
+                            // First iteration must use where instead of orWhere, as seen in filterWorkRegions
+                            $query->where($equityVars[$index], true);
+                        } else {
+                            $query->orWhere($equityVars[$index], true);
+                        }
+                    }
+                });
         });
         return $query;
     }
 
+    public function filterByGeneralSearch(Builder $query, ?string $search): Builder
+    {
+        // used App\\Models\\User@filterByPools as reference
+        if ($search) {
+            $query->whereExists(function ($query) use ($search) {
+                $query->select('id')
+                    ->from('users')
+                    ->whereColumn('users.id', 'pool_candidates.user_id')
+                    ->where(function ($query) use ($search) {
+                        $query->where('users.first_name', "ilike", $search)
```

Comment:
should be  `"%{$search}%"`

Reply:
done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/4129/commits/1776e91b9c01c0645193ba4c446f2632bdc26923

Reply:
🙏